static/src/helpers/utils.ts

```diff
@@ -186,3 +187,32 @@ export async function sendDeactivatingCallback(router: HTMLIonRouterOutletElemen
     el.deactivatingCallback();
   }
 }
+
+export function preProcessMultilineText(text: string) {
+  // This function exists to handle backward compatibility with older versions.
+  // Before the introduction of the WYSIWYG edtior,
+  // all text was rendered as the original text with whitespace preserved.
+
+  // Return early if there are no newlines in the text.
+  if (!text.includes('\n') && !text.includes('\r')) {
+    return text;
+  }
+
+  console.log('original', text);
+
+  // Convert all newlines to <br> tags.
+  text = text.replace(/(\r\n|\r|\n)/g, '<br>');
+  // Preserve all consecutive spaces.
+  text = text.replace(/\s/gm, '&nbsp;');
```

Comment:
Need to use function to process match so we can replace each space with alternating &nbsp; and 'real space so that link breaks work.